web/src/app/administration/components/workbasket-list-toolbar/workbasket-list-toolbar.component.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { WorkbasketState } from '../../../shared/store/workbasket-store/workbasket.state';
+import { WorkbasketService } from '../../../shared/services/workbasket/workbasket.service';
+import { DomainService } from '../../../shared/services/domain/domain.service';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { MatDialogModule } from '@angular/material/dialog';
+import { CreateWorkbasket } from '../../../shared/store/workbasket-store/workbasket.actions';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { Filter } from '../../../shared/models/filter';
+import { Sorting } from '../../../shared/models/sorting';
+import { ACTION } from '../../../shared/models/action';
+import { TaskanaType } from '../../../shared/models/taskana-type';
+
+const getDomainFn = jest.fn().mockReturnValue(true);
+const domainServiceMock = jest.fn().mockImplementation(
+  (): Partial<DomainService> => ({
+    getDomains: getDomainFn
+  })
+);
+
+@Component({ selector: 'taskana-administration-import-export', template: '' })
+class ImportExportStub {
+  @Input() currentSelection: TaskanaType;
+}
+
+@Component({ selector: 'taskana-shared-sort', template: '' })
+class SortStub {
+  @Input() sortingFields: Map<string, string>;
+  @Input() defaultSortBy = 'key';
+  @Output() performSorting = new EventEmitter<Sorting>();
+}
+
+@Component({ selector: 'taskana-shared-filter', template: '' })
+class FilterStub {
+  @Output() performFilter = new EventEmitter<Filter>();
+}
+
+describe('WorkbasketListToolbarComponent', () => {
+  let fixture: ComponentFixture<WorkbasketListToolbarComponent>;
+  let debugElement: DebugElement;
+  let component: WorkbasketListToolbarComponent;
+  let store: Store;
+  let actions$: Observable<any>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        NgxsModule.forRoot([WorkbasketState]),
+        MatSnackBarModule,
+        MatDialogModule,
+        BrowserAnimationsModule
+      ],
+      declarations: [WorkbasketListToolbarComponent, ImportExportStub, SortStub, FilterStub],
+      providers: [{ provide: DomainService, useClass: domainServiceMock }, WorkbasketService]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(WorkbasketListToolbarComponent);
+    debugElement = fixture.debugElement;
+    component = fixture.debugElement.componentInstance;
+    store = TestBed.inject(Store);
+    actions$ = TestBed.inject(Actions);
+    component.action = ACTION.COPY;
+  }));
+
+  it('should create the component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should dispatch CreateWorkbasket when addWorkbasket is called', async((done) => {
+    component.addWorkbasket();
+    let actionDispatched = false;
+    actions$.pipe(ofActionDispatched(CreateWorkbasket)).subscribe(() => {
+      actionDispatched = true;
+      expect(actionDispatched).toBe(true);
```

Comment:
moved expect out of subscription, this subscription shouldn't be called so that actionDispatched should always be false. Expect false at the end of test case will still be checked

Reply:
You are setting actionDispatched to true right before you test whether actionDispatched is true. Therefore, the test will never fail.
However, this test should fail when the action is not dispatched although the action is not CREATE. But the test won't fail.

Reply:
I think the expect should be outside of the subscription.